projectDocs/dev/deprecations.md

```diff
@@ -27,16 +27,18 @@ def __getattr__(attrName: str) -> Any:
 ```
 
 ## Required API breaking changes
+
 In order to improve the NVDA API, changes that will break future compatibility may be implemented, as long they retain backwards compatibility until the `20XX.1` release.
 
 This can be done by using a version check to automate deprecation.
 For example, if you wish to replace usages of `deprecatedSymbolName` with `newSymbolName`.
-When we begin work on `NEXT_YEAR`, `deprecatedSymbolName` will no longer be part of the NVDA API and all internal usages must be removed prior.
+When we begin work on `NEXT_YEAR`, we update `BACK_COMPAT_TO`, which introduces the add-on API breakage warning.
+At this stage, `deprecatedSymbolName` will no longer be part of the NVDA API and all internal usages must be removed prior.
 
 ```python
-from buildVersion import version_year
+from addonAPIversion import BACK_COMPAT_TO
 import NVDAState
-if version_year < NEXT_YEAR and NVDAState._allowDeprecatedAPI():
+if BACK_COMPAT_TO < (NEXT_YEAR, 1, 0) and NVDAState._allowDeprecatedAPI():
```

Comment:
This is generally how we use `BACK_COMPAT_TO`, we generally rely on tuple comparisons for version checks in general in NVDA.

I think comparing the full version is better, as it more accurately describes the process.
The same comparison method add-ons are using should be used to determine whether or not we use the new API. We are indeed relying on the fact that BACK_COMPAT_TO is before the full API version, not just the year. The 20XX.1 rule exists outside of code, so it's not logic I think we should be relying on. For example, if we have an unscheduled/unusual API breaking release, having the process generic enough is ideal. The 20XX.1 rule for API breaking releases may need to be broken now and then, we've skirted pretty close before with having to perform major security changes in a non-API breaking release cycle.

Could you explain more what you think is unintuitive? Even without prior knowledge that tuples are compared item-by-item in order, I think it can be inferred what this code does, particularly with the accompanying explanation.

```py
if BACK_COMPAT_TO < (2025, 1, 0) and NVDAState._allowDeprecatedAPI():
```

Reply:
The issue with using a NamedTuple is that changing the type of `AddonApiVersionT` would have fairly sweeping applications and require  a thorough refactor. I think it's out of scope of this PR but should be considered in future.

Reply:
On further consideration, I don't think it's unintuitive. I blame the fact that I hadn't had any coffee this morning.